### ISSUE TITLE
Retrieving list of corpora from Sketch Engine not working

### ISSUE THREAD

**Issue by iztokkosem** [elexis-eu/lexonomy#165]
On our local installation we are having a problem with Sketch Engine connection, specifically with getting a list of available corpora. The message "Retrieving available corpora from Sketch Engine, please wait..." never disappears, the list is never obtained.

I don't know whether this is associated with my username in particular, although it should not be as I'm using the same one for lexonomy.eu. Also, it works on one existing dictionary where the designated corpus was manually put in before this "listing" feature was introduced.

**Comment by gareins** [comment]
It seems to me, that there are CORS headers missing/missconfigured for SKE api call directly from javascript, which is performed in kex.js:33 . I have updated to latest master and no update fixed this.

Firefox says: Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://api.sketchengine.eu/ca/api/corpora. (Reason: CORS header 'Access-Control-Allow-Origin' missing).

Simmilar message is shown by Chrome.

**Comment by mjakubicek** [comment]
yes, you are right, this needs to go through the Lexonomy backend to be CORS compliant. If you would be doing that, please do that in the beta branch, not master.

**Comment by gareins** [comment]
I see that backend /skeget/corpora appends the username, does that change the behaviour? If not, then you would probably like this kind of a solution: a9dd2da47b6b10921cff4303a8d0a1bc64604baa
Or maybe do you have some SKE documentation available for me to check in times such as these?

**Comment by mjakubicek** [comment]
nope nope, you miss the point here. It is about CORS compliance. We never send headers that would allow XHR calls from non-Sketch Engine website, that would be a huge security risk. That's why you need to route the call through Lexonomy backend.

**Comment by mjakubicek** [comment]
not to mention the fact that because credentials are transferred as well, even sending the headers would not help here because you cannot set A-C-A-O to * when sending credentials. Read about CORS for more details.

**Comment by gareins** [comment]
Many thanks for a quick reply!

I don't really see how your response applies to the proposed solution, but I was wrong before, so I had to check.

My first mistake: I assumed that this `skeget_corpora` function in `lexonomy.py` makes the same request as the kex widget, but it turns out that is not the case. Also, I think it actually has a bug (`username=` missing equal sign). So I actually did the work and made it forward correct headers and it works now. The code:

```py
@get(siteconfig["rootPath"] + "skeget/corpora")                                                                                       
def skeget_corpora():                                                                                                                 
    req = urllib.request.Request("https://api.sketchengine.eu/ca/api/corpora", 
       headers={'Authorization': request.headers['Authorization']})                                                                                                                                 
                                                                                                                                      
    ske_response = urllib.request.urlopen(req)                                                                                        
    response.headers['Content-Type'] = ske_response.getheader('Content-Type')                                                         
                                                                                                                                      
    return ske_response
 ```

Would this be appropriate solution (I would also rename `@get` path to not change the current code), or am I still doing the same dumb thing, not following CORS and other browser-security stuff?
